devito/types/basic.py

```diff
@@ -747,7 +746,7 @@ def __init__(self, *args, **kwargs):
             super(Array, self).__init__(*args, **kwargs)
 
             self._scope = kwargs.get('scope', 'heap')
-            assert self._scope in ['heap', 'stack', 'external']
+            assert self._scope in ['heap', 'stack']
```

Comment:
so external isn't allowed anymore?

Reply:
correct, it was just redundant -- over time, I got to the point that Arrays are allocated internally by Devito and they never are attached data, so they are implicitly 'external' (see also the inherited ``_mem_external`` property)